### PR TITLE
In App Notifications Support

### PR DESCRIPTION
**Pull request by yarneo**
In App Notifications Support for Swift3

### CODE REVIEW

File: Mixpanel/DecideRequest.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+class DecideRequest: Network {
+
+    typealias DecideResult = [String: AnyObject]
+    let decidePath = "/decide"
+    var networkRequestsAllowedAfterTime = 0.0
+    var networkConsecutiveFailures = 0
+
+    struct DecideQueryItems {
+        let version: URLQueryItem
```
Comment (patedit) [inline Mixpanel/DecideRequest.swift]: don't we also send `ios_app_version`, `ios_lib_version` and some more params on every decide request?

File: Mixpanel/DecideRequest.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    private func decideRequestHandler(_ base: String,
+                                      resource: Resource<DecideResult>,
+                                      completion: (DecideResult?) -> Void) {
+        Network.apiRequest(base: base,
+                           resource: resource,
+                           failure: { (reason, data, response) in
+                            Logger.warn(message: "API request to \(resource.path) has failed with reason \(reason)")
+                            completion(nil)
+            },
```
Comment (patedit) [inline Mixpanel/DecideRequest.swift]: formatting

File: Mixpanel/InAppNotifications.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+//  Mixpanel
+//
+//  Created by Yarden Eitan on 8/9/16.
+//  Copyright © 2016 Mixpanel. All rights reserved.
+//
+
+import Foundation
+
+protocol InAppNotificationsDelegate {
+    func markNotification(_ notification: InAppNotification)
```
Comment (patedit) [inline Mixpanel/InAppNotifications.swift]: The delegate does not have to mark a notification or track a notification. Rather, get notified that when that happened to do extra work. Do you think it makes sense to rename these to `onNotificationMarked` and `onNotificationTracked`?

File: Mixpanel/BaseNotificationViewController.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+//  Mixpanel
+//
+//  Created by Yarden Eitan on 8/11/16.
+//  Copyright © 2016 Mixpanel. All rights reserved.
+//
+
+import UIKit
+
+protocol NotificationViewControllerDelegate {
+    func dismissNotification(controller: BaseNotificationViewController, status: Bool)
```
Comment (patedit) [inline Mixpanel/BaseNotificationViewController.swift]: Same here

File: Mixpanel/InAppNotifications.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            controller.hide(animated: true) {
+                Logger.info(message: "opening CTA URL: \(URL)")
+
+                if !UIApplication.shared.openURL(URL) {
+                    Logger.error(message: "Mixpanel failed to open given URL: \(URL)")
+                }
+
+                self.delegate?.trackNotification(controller.notification, event: "$campaign_open")
+                completionBlock()
+
```
Comment (patedit) [inline Mixpanel/InAppNotifications.swift]: space

File: Mixpanel/DecideRequest.swift (hunk 1 above)
Comment (yarneo) [inline Mixpanel/DecideRequest.swift]: They are in the properties variable, have a look at line 31 :)

**Pull request merged**